### PR TITLE
Chore: use exexec from upstream

### PR DESCRIPTION
**Pull request by ayrat555**
fixes https://github.com/omgnetwork/elixir-omg/issues/1616

### CODE REVIEW

File: apps/omg_eth/test/support/dev_geth.ex [hunk 1]
```diff
@@ -36,14 +36,13 @@ defmodule OMG.Eth.DevGeth do
     geth = ~s(geth --miner.gastarget 7500000 \
             --nodiscover \
             --maxpeers 0 \
-            --miner.gasprice \"10\" \
-            --datadir /data/ \
```
Comment (ayrat555) [inline apps/omg_eth/test/support/dev_geth.ex:40]: the second datadir is on line 50

File: apps/omg_eth/test/support/dev_geth.ex [hunk 2]
```diff
@@ -36,27 +38,75 @@ defmodule OMG.Eth.DevGeth do
...
+    GenServer.call(pid, :ready?)
+  end
+
+  def stop_geth(pid) do
+    GenServer.cast(pid, :stop)
+  end
+
+  def start_link(cmd) do
+    GenServer.start_link(__MODULE__, cmd)
+  end
```
Comment (InoMurko) [inline apps/omg_eth/test/support/dev_geth.ex:71]: these could be private?

Comment (ayrat555) [reply apps/omg_eth/test/support/dev_geth.ex]: @InoMurko I made them private and fixed tests

**Comment by ayrat555** [comment]
@InoMurko @pgebal does it look good?

File: apps/omg_eth/test/support/dev_geth.ex [hunk 3]
```diff
@@ -36,54 +38,88 @@ defmodule OMG.Eth.DevGeth do
...
-      end
+    pid
+  end
+
+  defp wait_for_rpc(pid) do
+    if ready?(pid) do
+      :ok
+    else
+      Process.sleep(1_000)
+      ready?(pid)
```
Comment (pgebal) [inline apps/omg_eth/test/support/dev_geth.ex:114]: shouldn't it be `wait_for_rpc`?

**Review by pgebal — CHANGES_REQUESTED** [review]

File: apps/omg_eth/test/support/dev_geth.ex [hunk 4]
```diff
@@ -36,54 +38,88 @@ defmodule OMG.Eth.DevGeth do
...
+
+  # PRIVATE
+
   defp launch(cmd) do
-    _ = Logger.debug("Starting geth")
+    {:ok, pid} = start_link(cmd)
 
-    {:ok, geth_proc, _ref, [{:stream, geth_out, _stream_server}]} =
-      Exexec.run(cmd, stdout: :stream, kill_command: "pkill -9 geth")
+    waiting_task_function = fn ->
```
Comment (pgebal) [inline apps/omg_eth/test/support/dev_geth.ex:98]: I think `function` part in var name in unnecessary. We can take a look on the right side of the assignment and we know it's a function.

**Review by pgebal — CHANGES_REQUESTED** [review]

File: apps/omg_eth/test/support/dev_geth.ex (hunk 3 above)
Comment (ayrat555) [reply apps/omg_eth/test/support/dev_geth.ex]: you're right

File: apps/omg_eth/test/support/dev_geth.ex [hunk 5]
```diff
@@ -36,54 +38,88 @@ defmodule OMG.Eth.DevGeth do
...
+  @impl true
+  def handle_info({:stdout, pid, stdout}, %{os_proc: pid} = state) do
+    new_state =
+      if String.contains?(stdout, "IPC endpoint opened") do
+        Map.put(state, :ready?, true)
+      else
+        state
+      end
+
+    if Application.get_env(:omg_eth, :node_logging_in_debug) do
```
Comment (InoMurko) [inline apps/omg_eth/test/support/dev_geth.ex:81]: Not your doing, but this should be a case.

**Review by InoMurko — APPROVED** [review]
I guess same goes into omg childchain v1!

**Review by pgebal — APPROVED** [review]

**Pull request merged**